lib/editor-settings.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Utilities to manage editor settings.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Returns editor settings that are common to all editors:
+ * post, site, widgets, and navigation.
+ *
+ * All these settings are already part of core,
+ * see edit-form-blocks.php
+ *
+ * @return array Common editor settings.
+ */
+function gutenberg_get_common_block_editor_settings() {
```

Comment:
Interesting, maybe it should be refactored to work as a filter that is applied everywhere but not to the `edit-post` page? This way, it might be simpler to backport it to the WordPress core. It's optional though so I leave it up to you to decide.

Reply:
Right! I plan to iterate on this towards the direction you suggest, but I'm unsure how at the moment. I presume all editors should have extensibility points but I need to look at this deeply to know what that means in practice. For example, should all editors run the `block_editor_settings` filter or should they have separate ones? This is the same as asking: is the consumer's responsibility to know which page are we loading (so we can use the existing `block_editor_settings`) VS should consumers be able to hook into the specific editors instead. It's a different conversation that merits its own PR.